kinisi/jump_diffusion_analyzer.py

```diff
@@ -31,218 +34,69 @@ class JumpDiffusionAnalyzer(Analyzer):
-        jdiff_anal._diff = diffusion.MSTDBootstrap(jdiff_anal._delta_t, jdiff_anal._disp_3d, jdiff_anal._n_o,
-                                                   **uncertainty_params)
-        return jdiff_anal
-
-    def jump_diffusion(self, start_dt: float, jump_diffusion_params: Union[dict, None] = None):
-        """
-        Calculate the jump diffusion coefficicent using the bootstrap-GLS methodology.
-
-        :param start_dt: The starting time for the analysis to find the diffusion coefficient.
-            This should be the start of the diffusive regime in the simulation.
-        :param ump_diffusion_params: The parameters for the :py:class:`kinisi.diffusion.MSTDBootstrap`
-            object. See the appropriate documentation for more guidance on this. Optional, default is the
-            default bootstrap parameters.
-        """
-        if jump_diffusion_params is None:
-            jump_diffusion_params = {}
-        self._diff.jump_diffusion(start_dt, **jump_diffusion_params)
-
-    @property
-    def mstd(self) -> np.ndarray:
-        """
-        :return: MSTD for the input trajectories. Note that this is the bootstrap sampled MSD, not the numerical
-            average from the data.
+            then it is necessary to identify if these constitute a series of :py:attr:`consecutive` trajectories or
+            a series of :py:attr:`identical` starting points with different random seeds, in which case the `dtype`
+            should be either :py:attr:`consecutive` or :py:attr:`identical`.:
+        :param dt: Time intervals to calculate the displacements over. Optional, defaults to a :py:mod:`scipp` array
+            ranging from the smallest interval (i.e., time_step * step_skip) to the full simulation length, with 
+            a step size the same as the smallest interval.
+        :param dimension: Dimension/s to find the displacement along, this should be some subset of `'xyz'` indicating
+            the axes of interest. Optional, defaults to `'xyz'`.
+        :param distance_unit: The unit of distance in the simulation input. This should be a :py:mod:`scipp` unit and
+            defaults to :py:attr:`sc.units.angstrom`.
+        :param progress: Print progress bars to screen. Optional, defaults to :py:attr:`True`.
+        
+        :returns: The :py:class:`DiffusionAnalyzer` object with the mean-squared displacement calculated.
         """
-        return self._diff.n
-
+        p = super()._from_Xdatcar(trajectory, specie, time_step, step_skip, dtype, dt, dimension, distance_unit, progress)
+        p.mstd = calculate_mstd(p.trajectory, progress)
+        return p
+    
     @property
-    def mstd_std(self) -> np.ndarray:
-        """
-        :return: MSD standard deviation values for the input trajectories (a single standard deviation).
-        """
-        return self._diff.s
+    def n_atoms(self):
+        """Property to access the n_atoms."""
+        return n_atoms(self.trajectory)
```

Comment:
```suggestion
        return sc.scalar(self.trajectory.displacements.sizes['atom'])
```

Does something like this work?

Reply:
This way you can delete the `n_atoms` function from the `displacments.py` module.

Reply:
Makes sense, much less clunky